utils/utils.go

```diff
@@ -248,6 +248,18 @@ func GeneratePasswd(uid, gid uint32, rootfs, rundir string) (string, error) {
 	if err != nil {
 		return "", errors.Wrapf(err, "unable to follow symlinks to passwd file")
 	}
+	info, err := os.Stat(originPasswdFile)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return "", nil
+		}
+		return "", errors.Wrapf(err, "unable to stat passwd file %s", originPasswdFile)
+	}
+	// Check if passwd file is world writable
+	if info.Mode().Perm()&(0022) != 0 {
+		return "", nil
+	}
+
```

Comment:
We need another check for skipping for u+w. So, if uid matches the file owner's uid, then we can skip.

Reply:
Added